### PR TITLE
Nmi notificaton for v1.5

### PR DESCRIPTION
**Pull request by KaigeFu**

### CODE REVIEW

**Comment by KaigeFu** [comment]
@wenlingz, nmi_notification for v1.5 PRed.

**Comment by HuiHuangShi** [comment]
6 new potential violations to the coding guideline detected. Please help check. Thanks!
<1> config:sharing_config filename:/hypervisor/arch/x86/notify.c func:is_notification_nmi offset:125
reason:Pointer param should be declared pointer to const. : vm
<2> config:part_config filename:/hypervisor/arch/x86/notify.c func:is_notification_nmi offset:125
reason:Pointer param should be declared pointer to const. : vm
<3> config:common filename:/hypervisor/include/arch/x86/guest/vm.h func:vcpu_from_vid offset:171
reason:Static procedure is not explicitly called in code analysed. : vcpu_from_vid
<4> config:common filename:/hypervisor/arch/x86/guest/virq.c func:nmi_window_vmexit_handler offset:535
reason:Basic type declaration used. : int nmi_window_vmexit_handler
<5> config:sharing_config filename:/hypervisor/arch/x86/guest/virq.c func:nmi_window_vmexit_handler offset:535
reason:Procedure is not called or referenced in code analysed. : nmi_window_vmexit_handler
<6> config:part_config filename:/hypervisor/arch/x86/guest/virq.c func:nmi_window_vmexit_handler offset:535
reason:Procedure is not called or referenced in code analysed. : nmi_window_vmexit_handler

No New Name Conflict

Function Declaration/Definition Mismatch Violation:
extern	int32_t nmi_window_vmexit_handler(struct acrn_vcpu * vcpu)	212	include/arch/x86/irq.h
define	int32_t nmi_window_vmexit_handler(struct acrn_vcpu * __attribute__((unused)) vcpu)	535	arch/x86/guest/virq.c

File: hypervisor/include/arch/x86/irq.h [hunk 1]
```diff
@@ -207,6 +209,7 @@ void vcpu_make_request(struct acrn_vcpu *vcpu, uint16_t eventid);
  * @pre vcpu != NULL
  */
 int32_t exception_vmexit_handler(struct acrn_vcpu *vcpu);
+int32_t nmi_window_vmexit_handler(struct acrn_vcpu *vcpu);
```
Comment (HuiHuangShi) [inline hypervisor/include/arch/x86/irq.h]: please add "__unused" prefix to keep the consistency.

Comment (KaigeFu) [reply hypervisor/include/arch/x86/irq.h]: @HuiHuangShi. Actually, we use the parameter "vcpu" in nmi_window_vmexit_handler. But I add the __unused prefix in its implementation by mistake. I have fix it in next version.

**Comment by HuiHuangShi** [comment]
3 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/include/arch/x86/guest/vm.h func:vcpu_from_vid offset:171<br> reason:Static procedure is not explicitly called in code analysed. : vcpu_from_vid<br><2> config:sharing_config filename:/hypervisor/arch/x86/guest/virq.c func:nmi_window_vmexit_handler offset:535<br> reason:Procedure is not called or referenced in code analysed. : nmi_window_vmexit_handler<br><3> config:part_config filename:/hypervisor/arch/x86/guest/virq.c func:nmi_window_vmexit_handler offset:535<br> reason:Procedure is not called or referenced in code analysed. : nmi_window_vmexit_handler<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
3 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/include/arch/x86/guest/vm.h func:vcpu_from_vid offset:171<br> reason:Static procedure is not explicitly called in code analysed. : vcpu_from_vid<br><2> config:sharing_config filename:/hypervisor/arch/x86/guest/virq.c func:nmi_window_vmexit_handler offset:535<br> reason:Procedure is not called or referenced in code analysed. : nmi_window_vmexit_handler<br><3> config:part_config filename:/hypervisor/arch/x86/guest/virq.c func:nmi_window_vmexit_handler offset:535<br> reason:Procedure is not called or referenced in code analysed. : nmi_window_vmexit_handler<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by KaigeFu** [comment]
@HuiHuangShi suppression table updated.

**Pull request merged**